src/shared/logs-show.c

```diff
@@ -403,24 +403,25 @@ static int output_timestamp_realtime(
                         break;
 
                 case OUTPUT_SHORT_ISO:
-                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S%z",
-                                     localtime_or_gmtime_r(&t, &tm, flags & OUTPUT_UTC)) <= 0)
+                case OUTPUT_SHORT_ISO_PRECISE: {
+                        size_t tail = strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S",
+                                               localtime_or_gmtime_r(&t, &tm, flags & OUTPUT_UTC));
+                        if (tail == 0)
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Failed to format ISO time");
-                        break;
 
-                case OUTPUT_SHORT_ISO_PRECISE: {
-                        char usec[7];
+                        /* No usec in strftime, need to append */
+                        if (mode == OUTPUT_SHORT_ISO_PRECISE) {
+                                snprintf(buf + tail, ELEMENTSOF(buf) - tail, ".%06"PRI_USEC, display_ts->realtime % USEC_PER_SEC);
+                                tail += 7;
+                        }
 
-                        /* No usec in strftime, so we leave space and copy over */
-                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.xxxxxx%z",
-                                     localtime_or_gmtime_r(&t, &tm, flags & OUTPUT_UTC)) <= 0)
-                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                       "Failed to format ISO-precise time");
-                        xsprintf(usec, "%06"PRI_USEC, display_ts->realtime % USEC_PER_SEC);
-                        memcpy(buf + 20, usec, 6);
+                        int8_t h = tm.tm_gmtoff / 60 / 60;
+                        int8_t m = labs((tm.tm_gmtoff / 60) % 60);
```

Comment:
hmm, where does the `int8_t` type come from? this is so weird, a signed byte? we have never used such a type, as when put in registers or in the stack is upgraded to `int` anyway.

moreover tm_gmtoff actually uses `long` as type, hence the artificially shorter width is just weird and smeels a lot like accidental loss of precision. 

Quite frankly the compiler should warn assigning the result of a funciton like labs() that operates on long ints onto an int8_t...

Reply:
From stdint.h, like all the others. If I had a type that went [0, 60) I would've used it, but this uses int8_t because it's derived from voreutils vore-time which carries a generic version of date(1) %z, but there it's
```cpp
							std::int8_t hms[3];
							hms[2] = std::abs(tm_gmtoff_for(tm, ts) % 60);
							hms[1] = std::abs((tm_gmtoff_for(tm, ts) / 60) % 60);
							hms[0] = tm_gmtoff_for(tm, ts) / 60 / 60;
```